frameworks/AutoGluon/exec.py

```diff
@@ -48,6 +48,17 @@ def run(dataset, config):
 
     is_classification = config.type == 'classification'
     training_params = {k: v for k, v in config.framework_params.items() if not k.startswith('_')}
+    if "high_quality" in training_params.get("presets", []):
```

Comment:
You need to first check if `presets` is not a list, if it is not a list, need to convert to list, otherwise this won't work correctly.

```python3
presets = training_params.get("presets", [])
if not isinstance(presets, list) and presets is not None:
    presets = [presets]
if presets is not None and ("high_quality" in presets or "good_quality" in presets):
    preserve = 0.9
    ...
```

Reply:
Thanks! Updated.

Reply:
LGTM!